terraform/environments/data-platform/iam.tf

```
@@ -1,5 +1,35 @@
 # IAM policy documents for lambda functions
 
+data "aws_iam_policy_document" "log_to_bucket" {
+  statement {
+    sid    = "s3LogAccess"
+    effect = "Allow"
+    actions = [
+      "s3:GetObject",
+      "s3:PutObject",
```

```suggestion
      "s3:PutObject",
      "s3:ListBucket",
      "s3:GetBucketLocation"
```
Does we want to add log bucket visibility/discovery to this statement?

---

Nothing needs to list the contents of the log bucket at the moment, so Get and Put is enough. So I was going to leave this out and just go with the principle of least privilege. If you think we might need it I'm happy to add it in though.

If we add an endpoint for fetching the logs, we might need ListBucket for that, but I think we can handle that separately.

---

Yes it looks like I was mistaken - I think this might be a console requirement, not a role requirement for accessing items within a bucket.